### ISSUE TITLE
I can't recognize picture tags with saucenao\iqdb

### ISSUE THREAD

**Issue by mehmeh1z** [tauraloke/tegownitsa#99]
And speaking of bugs, I ran into the fact that the tags for the saucenao\iqdb picture are not assigned. And it would be a little-known picture, but I specifically took from the same danbooru, and tags just did not appear.
(Although in the very first run of the application, i was able to pull the tags for the iqdb\saucenao picture, but after that, I never got around to doing it again.)

**Comment by tauraloke** [comment]
The most obvious reasons of the same behavior I have in my environment were:
1) the host of a remote search engine is temporarily unavailable.
2) the host of a booru with tags is temporarily unavailable.
3) I had retrieved tags after a totally bad tag run and the picture has blocking marks (special marks are stopping repeated tag retrieving to reduce traffic; but you can free remove this blocking marks in the picture screen from the block "Pollee tag sources").

Sometimes reasons was more exotic due to bugs in program logic. So, I looked through the logs by Ctrl+Shift+I and got some additional info.

**Comment by mehmeh1z** [comment]
Am I doing it wrong?
For example, I add a picture => Edit =>Load tags from Saucenao for current files => Predict tags by neuronet
And one time the tags themselves were added to the picture, all the other times I had to add each tag manually.

**Comment by tauraloke** [comment]
First, you can check Saucenao API key in Edit / Settings (Ctrl+,) / Tag sources / Saucenao. Then you can check your daily limit on the same screen.
BTW IQDB do not request api key so I don't know what exactly happened.

Neuronet prediction is a separate options. If you have already downloaded the model files, you can predict tags even if booru don't know about the image. It is usually a bit long procedure the first time.

**Comment by mehmeh1z** [comment]
When i check saucenao daily limits:
30s: 0 (limit 4)
24h: 0 (limit 100)
When i try to load tags from saucenao after add api:
All files in query are already inspected for tags

**Comment by tauraloke** [comment]
Looks like unused free limits. So you can still use Saucenao with it.
You can remove the pollee mark from the picture after the unsuccess attempt (in the picture screen) and try get tags again.

**Comment by mehmeh1z** [comment]
I tried it, but nothing changed.
The tags were not added.

**Comment by tauraloke** [comment]
Yes, it's strange. If you remove pollee marks of boorus and search engines from the picture, you can still get tags.
What is output to the console in developer tools (Ctrl+Shift+I)?

**Comment by mehmeh1z** [comment]
I think I got it.
Maybe the problem is that it's trying to take tags from pixiv (that's the topmost source links in saucenao), and it can't do it.
![изображение](https://user-images.githubusercontent.com/55963243/204747694-172cf370-4503-48f9-b876-a830ccd28129.png)

**Comment by mehmeh1z** [comment]
That's it, now I think I've finally figured it out.
Started vpn and tried again to get tags through saucenao and everything was successful.
Maybe there's a way to add ISP blocking bypass/proxy to your program?

**Comment by tauraloke** [comment]
So it's case #2 (a booru is unavailable).

> I guess it probably works through chrome (considering the kind of console in your program).
It's Electron framework with Chromium browser engine. Full add-ons are not supported: https://www.electronjs.org/docs/latest/api/extensions

> Maybe there's a way to add ISP blocking bypass/proxy to your program?
I think it can be solved with low-level packages like https://www.npmjs.com/package/https-proxy-agent
It's not a simple way ofcource.

I have added proxy options to app roadmap.

**Comment by mehmeh1z** [comment]
>I have added proxy options to the app roadmap.

Would it be possible to add a url field to enter automatic proxy settings? (like in firefox)
